### DISCUSSION TITLE
Проблема с комментариями

### DISCUSSION THREAD

**Discussion by Vadim017** [totumonline/totum-ru-issues-and-discussions#619] [Как это сделать?] [answered]
Мне необходимо, чтоб коменты из временной таблицы записывались в простую таблицу. Таблицы одинаковые, всё прописано вроде бы правильно и от администратора работает во всех таблицах. Таких парных таблиц у меня 5, 5 простых таблиц в которых все данные и 5 временных таблиц для отображения части информации. От учётки пользователя работает только в одной из таблиц, права доступа для всех таблиц и столбцов прописаны одинаково во всех. Для роли пользователя все таблицы открыты в полном доступе. Код для экспорта комментариев из простой во временную таблицы работает отлично. Прикреплённый код, это код действия по триггеру изменение во временной таблице, для изменения списка комментариев в простой таблице.
[код для комментов.txt](https://github.com/user-attachments/files/17646742/default.txt)

**Comment by Vadim017** [comment]
Буду очень благодарен, если кто нибудь поможет, я уже с этой проблемой вторую неделю не могу разобраться)

**Comment by ttm-alexeypolunin** [comment]
Сам код нормальный (только вот так пожалуйста):

```totum
=: setListExtended(table: 'tablica_ucheta_ecp'; field: 'kommentarii' = $fio[[kommentarii]]; where: 'id' = $fio[[id_nom]]; log: true)

	~fio: selectRowList(table: $#ntn; field: 'id_nom'; field: 'kommentarii')
```

Включите логгирование, переключитесь в sudo-режиме в пользователя и выполните и посмотрите что происходит.

**Reply by Vadim017** [reply]
Доброе утро, если правильно понял, то сделал то, что вы сказали и получилось, что в таблице где комменты работают идёт в конце пересчёт конечной таблицы, 
![image](https://github.com/user-attachments/assets/4978c1f9-44df-4c4f-bf82-eec11790e250)
![не рабочие комменты](https://github.com/user-attachments/assets/ba96b5c6-cdfb-4085-95db-7f965b4c2b33)
![рабочие комменты](https://github.com/user-attachments/assets/3a972e04-2aab-41f1-9513-ab172b89264e)
 а в таблице где комменты не работают он не идёт. Подскажите как это исправить и где?

**Comment by Vadim017** [comment]
Если под администратором комментарий оставлять то всё работает
![image](https://github.com/user-attachments/assets/8b551989-5434-4bf4-b859-168e3f0bb603)
пересчёт внизу появляется, и появляется ch1 то есть действие по триггеру изменение

**Reply by ttm-alexeypolunin** [reply, marked as answer]
```
fio: [{"id_nom":null,"kommentarii":[["2024-11-07 10:02",144,"test"]]}]
```
id_nom: null — в той таблице в которой не работает поле с номерами пустое

**Reply by Vadim017** [reply]
Большое спасибо, я только сейчас понял в чём прикол, я сделал разные кнопки для администратора и пользователя и в кнопке для пользователя забыл прописать вызов id))))))
